microservices/src/main/java/io/silverware/microservices/silver/services/lookup/RoundRobinLookupStrategy.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * -----------------------------------------------------------------------\
+ * SilverWare
+ *  
+ * Copyright (C) 2010 - 2016 the original author or authors.
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------/
+ */
+package io.silverware.microservices.silver.services.lookup;
+
+import java.util.Set;
+
+/**
+ * Round robin choose which implementation is used in circular order.
+ *
+ * @author Slavomír Krupa (slavomir.krupa@gmail.com)
+ */
+public class RoundRobinLookupStrategy extends AbstractLookupStrategy {
+
+   private int index;
+
+   @Override
+   public Object getService() {
+      Set<Object> microservices = context.lookupMicroservice(metaData);
+      if (microservices.isEmpty()) {
+         throw new RuntimeException("No service found for: " + metaData);
+      }
+      Object[] servicesArray = microservices.toArray();
+      index++;
+      if (Integer.MAX_VALUE == index) {
+         index = 0;
+      }
+      return servicesArray[index % servicesArray.length];
```

Comment:
I think this method can be called from different threads at the same time which can now cause problems (for example `ArrayIndexOutOfBoundsException` on line 40). At least the part where the counter is increased (lines 40-43) should be synchronized. Or maybe also line 44 should be included. @marvec, what do you think?